.travis.yml

```diff
@@ -105,87 +100,87 @@ jobs:
-  #   install:
-  #     # Install GCP SDK if not already cached
-  #     - if [ ! -d "$HOME/google-cloud-sdk/bin" ]; then rm -rf "$HOME/google-cloud-sdk"; curl https://sdk.cloud.google.com | bash > /dev/null; fi
-  #     - source $HOME/google-cloud-sdk/path.bash.inc
-  #     - gcloud version
-  #     - pip install -r tests/requirements.txt
-  #     - ./tests/e2e/helm-preinstall.sh
-  #   before_script:
-  #     - openssl aes-256-cbc -K $encrypted_1019fd6574a7_key -iv $encrypted_1019fd6574a7_iv -in tests/e2e/e2e-gcp-sa-key.json.enc -out tests/e2e/e2e-gcp-sa-key.json -d
-  #     - gcloud auth activate-service-account --key-file tests/e2e/e2e-gcp-sa-key.json
-  #     - gcloud auth configure-docker
-  #     - ./build/docker-deploy.sh
-  #     - ./tests/e2e/before-deployment-test-helm-cleanup.sh
-  #     - ./tests/e2e/before-deployment-test-extensions-cleanup.sh
-  #   script:
-  #     - ./tests/e2e/execute-tests.sh
-  #   after_script:
-  #     - ./tests/e2e/deployment-test-cleanup.sh
+  - stage: e2e-helm-deployment-test
+    name: E2E GCP helm deployment test
+    sudo: required
+    language: python
+    python:
+      - "3.8"
+    if: (branch = master OR type = pull_request) AND fork = false
+    cache:
+      directories:
+        - "$HOME/google-cloud-sdk/"
+    env:
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+      - PUBSUB_TOPIC="e2e_test_topic_${TRAVIS_BUILD_ID}"
+      - PUBSUB_SUBSCRIPTION="e2e_test_subscription_${TRAVIS_BUILD_ID}"
+      - IAM_SERVICE_ACCOUNT="e2e-test-sa-${TRAVIS_BUILD_ID}"
+      - LOG_ROUTER="e2e_test_log_router_${TRAVIS_BUILD_ID}"
+      - IAM_ROLE_PREFIX="e2e_test_${TRAVIS_BUILD_ID}"
+      - K8S_CLUSTER="e2e-test-master"
+      - GCP_PROJECT_ID="dynatrace-gcp-extension"
+      - GCR_NAME="gcr.io/${GCP_PROJECT_ID}/dynatrace-gcp-function"
+      - E2E=true
+      - CLOUD_FUNCTION_NAME="e2e_test_sample_app"
+      - secure: "muHGNxIK6fxTltIsGgms6v3dleuQ1LZyQmZIGQDSfQ5sfYDYEvR2xsabK237AipSSGppoVhc7S6ri8A/WNQautHUfdBOsVn1i7czzoPhELdOJkAiigXrI3kla68OTCiC0TPwWbIYOxzmiYDHZr+rYbA1773O0iyhWHiTVQerLMndn0wCFWUUnLbD1WfzdAxlfSo1rd7u81m8g4efLIUppo/fjfaxmVwvo1oIfensGVknyOE+gtSh/OpGdYfNeOqxoDGOoMe5bnCqibUoREXRzUZKGb6upxMMmW6E3hGG02kqle4jYjaC7Uiu2IVZsK38iAWNGwIRy21CIXNHn4R6we44n5JLH92aiRtVw8D/fDOnN2gWTtmdrvoB8YklbaYZBnDrGJxI6rESfeQjwD8ImWWKUJFUlwovKjNWhOw8RMhPVtPOuC00+F2UDq5NosrWYGUjLjaKFC/4nysOiK2DgsMBF/Mk1Ei1BDMYmcHQ7Jsrs8ys/kgPasV3Qs9xOuSJpF+AiwKPfI1R9bxYiORg70TNz62wPD7FlKWIVe0DvhrWcHCTpnrROabt6B0hUROfljCZe2H+j8HD9uCJCJHk5qP7llNVofVnz8w+s90wLEKpUNn2rVUH0fzffYM7YaLRPhzLEUJSQso3a7hXFfcoUow8OOPTmCgViGCfPjuy2zc="
```

Comment:
I can see that those env variables are being reused in several stages. Wouldn't it be easier to store them under a single variable and call that variable instead?

Reply:
it duplicates i 2 sections but in one section this is different

Reply:
If those tokens are (or will be) subjects to change it might be worth moving them to the top of the pipeline and assign to env vars after all. This would make the code more readable... and DRY gods satisfied;)

Reply:
only one `- secure: "tdw5iV7XGhQ+gVPw70n+/SeA2xfVAXL5uwhMrKjI6F****` is in 2 stages but not needed on all stages so I don't see any advantages to move it to global